collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt

```diff
@@ -258,31 +247,42 @@ abstract class EntitiesRepositoryTest {
         repository.addList("blah")
         assertThat(repository.getLists(), containsInAnyOrder("wines", "blah"))
 
-        repository.save(Entity.New("wines", "blah", "Blah"))
+        repository.save("wines", Entity.New("blah", "Blah"))
         assertThat(repository.getLists(), containsInAnyOrder("wines", "blah"))
     }
 
     @Test
-    fun `#save supports properties with dots and dashes`() {
+    fun `#save supports properties with dots and dashes when saving new entities and updating existing ones`() {
         val repository = buildSubject()
         val entity = Entity.New(
-            "things",
             "1",
             "One",
             properties = listOf(Pair("a.property", "value"), Pair("a-property", "value"))
         )
-        repository.save(entity)
-        assertThat(repository.getEntities("things")[0], sameEntityAs(entity))
+
+        repository.save("things", entity)
+        val savedEntity = repository.getEntities("things")[0]
+        assertThat(savedEntity, sameEntityAs(entity))
+
+        repository.save("things", savedEntity)
+        assertThat(repository.getEntities("things")[0], sameEntityAs(savedEntity))
+    }
+
+    @Test
+    fun `#save with no entities works`() {
```

Comment:
This `works` is pretty generic it could mean that there is no crash or it does nothing... I think a better name would be:
`#save does not create a list when no entities are provided`